docs/openapi-docs/src/main/scala/sttp/tapir/docs/openapi/schema/ObjectSchemasForEndpoints.scala

```diff
@@ -20,10 +20,31 @@ object ObjectSchemasForEndpoints {
     val discriminatorToOpenApi = new DiscriminatorToOpenApi(schemaReferences)
     val tschemaToOSchema = new TSchemaToOSchema(schemaReferences, discriminatorToOpenApi)
     val schemas = new ObjectSchemas(tschemaToOSchema, schemaReferences)
-    val infosToSchema = sObjects.map(td => (td._1, tschemaToOSchema(td._2))).toMap
+    val infosToSchema = sObjects.map(td => (td._1, tschemaToOSchema(td._2))).toListMap
 
-    val schemaKeys = infosToSchema.map { case (k, v) => k -> ((infoToKey(k), v)) }
-    (schemaKeys.values.toListMap, schemas)
+    val schemaKeys = infosToSchema.map { case (k, v) => infoToKey(k) -> v }
+    (updateChildToParentRelation(schemaKeys), schemas)
+  }
+
+  private def updateChildToParentRelation(
```

Comment:
I'd add a comment explaining what it exactly does, and that it works only for coproducts (leaves others intact). And also why we need to add the backlink from the parent to the child :) (in a separate PR / directly to master)